ddtrace/settings/config.py

```diff
@@ -141,6 +159,229 @@ def get_error_ranges(error_range_str):
     return error_ranges  # type: ignore[return-value]
 
 
+_ConfigResult = TypedDict("_ConfigResult", {"value": Any, "raw": Any, "source": Any})
```

Comment:
Should we restrict the type of the configuration value? Using `Any` seems a bit too permissive